draft-ietf-privacypass-batched-tokens.md

```diff
@@ -395,18 +395,23 @@ struct {
 
 The structure fields are defined as follows:
 
-- "token_type" is a 2-octet integer. TokenRequest MUST be prefixed with a uint16
-  "token_type" indicating the token type. The rest of the structure follows
-  based on that type, within the inner opaque token_request attribute. The above
-  definition corresponds to TokenRequest from {{RFC9578}}. For TokenRequest not
-  defined in {{RFC9578}}, they MAY be used as long as they are prefixed with a
-  2-octet token_type.
-
-- "token_requests" are serialized TokenRequests, in network byte order. The
-  number of token_requests, as a 2-octet integer, is prepended to the serialized
-  TokenRequests. In addition, the 2-octet integer length of each TokenRequest is
-  prepended to the serialized TokenRequests.
+- TokenRequest's "token_type" is a 2-octet integer. TokenRequest MUST be prefixed
+  with a uint16 "token_type" indicating the token type. The rest of the
+  structure follows based on that type, within the inner opaque token_request
+  attribute. The above definition corresponds to TokenRequest from {{RFC9578}}.
+  For TokenRequest not defined in {{RFC9578}}, they MAY be used as long as they
+  are prefixed with a 2-octet token_type.
 
+- "token_requests" is an array of TokenRequest satisfying the above constraint.
+
+Note that when serialiazed in network byte order, BatchTokenRequest:
+
+- is prepended with a 2-octet integer representing the length of its underlying
+  byte array. This is not the number of TokenRequest.
+
+- has each of its TokenRequest serialized in network byte order. Specifically,
+  this implies each TokenRequest is prepended with its length as a 2-octet
+  integer.
```

Comment:
I'd remove this paragraph for now. Instead, there should be a section about TLS encoding elsewhere that applies to all structs with vectors.
As mentioned at IETF121, I think we should use [Variable-Size Vector Length Headers](https://www.rfc-editor.org/rfc/rfc9420.html#name-variable-size-vector-length).
I can follow up with a separate PR.

Reply:
having a dedicated paragraph about all struct would be good.
the structure uses fixed array as these were the ones used in the already written batched private tokens.
having variable tokens would work as well. the upside is to allow for more tokens (2^30 instead of 2^16), and save 1 byte when there are less than 2^6 tokens. the downside is it's harder to bound the possible privacy leak (especially with arbitrary batched tokens).

removing it for now